### PR TITLE
Add check for "project.json" and "extensions.csproj" when missing an assembly or reference in .csx

### LINKED ISSUE #4139: Warning on missing references
In .csx, we should also look for "project.json" (V1) or "extensions.csproj" (bindings) and emit a warning if those files are found.

### PR DESCRIPTION
**Pull request by mhoeger**
Resolves #4139

### CODE REVIEW

File: src/WebJobs.Script/Description/FunctionInvokerBase.cs [hunk 1]
```diff
@@ -162,6 +162,12 @@ internal void TraceCompilationDiagnostics(ImmutableArray<Diagnostic> diagnostics
             foreach (var diagnostic in diagnostics.Where(d => !d.IsSuppressed))
             {
                 FunctionLogger.Log(diagnostic.Severity.ToLogLevel(), 0, logState, null, (s, e) => diagnostic.ToString());
+                // Check if script compilation failed due to missing assembly (CS0246)
+                if (diagnostic.Id.Equals(DotNetConstants.TypeOrNamespaceNotFoundCompilerErrorCode))
```
Comment (fabiocav) [inline src/WebJobs.Script/Description/FunctionInvokerBase.cs]: This logic should be in the `DotNetFunctionInvoker`.

There's some existing logic that inspects compilation diagnostics and adds to the list of diagnostics. (look for `GetFunctionDiagnostics`)

**Comment by mhoeger** [comment]
@fabiocav - addressed CR comments! mind reviewing again?

**Comment by mhoeger** [comment]
@brettsam - mind checking this out?

File: src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs [hunk 2]
```diff
@@ -452,6 +458,25 @@ internal static object GetTaskResult(Task task)
...
+            if (File.Exists(deprecatedProjectPath))
+            {
+                string warning = string.Format(warningString, deprecatedProjectPath, DotNetConstants.ProjectFileName);
+                FunctionLogger.LogWarning(warning);
+            }
+            if (File.Exists(extensionsProjectPath))
+            {
+                string warning = string.Format(warningString, extensionsProjectPath, DotNetConstants.ProjectFileName);
+                FunctionLogger.LogWarning(warning);
+            }
```
Comment (brettsam) [inline src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs]: What if none of these files (nor the function.proj) are found? Can we write out something with the link to the docs in that case, too?

**Comment by mhoeger** [comment]
@brettsam - e2e tests added!

**Pull request merged**